libzkbob-rs-node/src/merkle.rs

```diff
@@ -209,15 +209,98 @@ pub fn merkle_get_virtual_node(mut cx: FunctionContext) -> JsResult<JsValue> {
     Ok(result)
 }
 
-pub fn merkle_rollback(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+pub fn merkle_rollback(mut cx: FunctionContext) -> JsResult<JsValue> {
     let tree = cx.argument::<BoxedMerkleTree>(0)?;
     let rollback_index = {
         let num = cx.argument::<JsNumber>(1)?;
         num.value(&mut cx) as u64
     };
 
     let mut a = tree.write().unwrap();
-    a.inner.rollback(rollback_index);
+    let new_next_index = a.inner.rollback(rollback_index);
+
+    let result = neon_serde::to_value(&mut cx, &new_next_index).unwrap();
+
+    Ok(result)
+}
+
+pub fn merkle_wipe(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+    let tree = cx.argument::<BoxedMerkleTree>(0)?;
+
+    tree.write().unwrap().inner.wipe();
 
     Ok(cx.undefined())
 }
+
+pub fn merkle_get_left_siblings(mut cx: FunctionContext) -> JsResult<JsValue> {
+    let tree = cx.argument::<BoxedMerkleTree>(0)?;
+    let index = {
+        let num = cx.argument::<JsNumber>(1)?;
+        num.value(&mut cx) as u64
+    };
+
+    if index & OUTPLUSONELOG as u64 != 0 {
+        return cx.throw_error(format!("Index to get sibling from should be multiple of {}", OUT + 1));
+    }
+
+    let siblings = tree.read().unwrap().inner.get_left_siblings(index);    
+    let array = match siblings {
+        Some(val) => {
+            let result: Result<Vec<String>, String> = val
+                .into_iter()
+                .map(|node| {
+                    let height = format!("{:#04x}", node.height);
+                    let index = format!("{:#014x}", node.index);
+
+                    let mut node_bytes = vec![];
+                    node.value.serialize(&mut node_bytes).unwrap();
+                    node_bytes.reverse();
+                    let node_string = hex::encode(node_bytes);
+
+
+                    let res = format!("{}{}{}",
+                        height.strip_prefix("0x").unwrap_or(&height),
+                        index.strip_prefix("0x").unwrap_or(&index),
```

Comment:
You can remove `#` flag when using `format!` macro to remove `strip_prefix` calls. `#` flag adds `0x` prefix which is not needed here.
https://doc.rust-lang.org/std/fmt/trait.LowerHex.html

Reply:
Agreed. Fixed in https://github.com/zkBob/libzkbob-rs/pull/33/commits/f6b7da07597f877d93d3f591f281e83045b6d1c1